scheduler/scheduler.go

```diff
@@ -29,9 +29,13 @@ type clonePair struct {
 	channel chan int
 }
 
+type Context interface {
```

Comment:
The name could be confused with context.Context which is a widely used Go type. If possible, we should find another name for this struct.